src/core/qgsauxiliarystorage.h

```diff
@@ -29,6 +30,60 @@
 class QgsProject;
 
 /**
+ * \class QgsAuxiliaryField
+ *
+ * \ingroup core
+ *
+ * \brief Class allowing to manage fields from of auxiliary layers
+ *
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsAuxiliaryField : public QgsField
```

Comment:
I'm not that comfortable with subclassing QgsField here. From my interpretation this class is only used as a convenience to creating QgsFields with the correct types/names, and after creation isn't used anywhere. In that case it could just be reduced to a static `QgsField createAuxiliaryField( const QgsPropertyDefinition &def)` method which returns a new QgsField with the correct type/name.

Reply:
It's also used to retrieve the underlying QgsPropertyDefinition in QgsVectorLayerProperties to build the widget tree in auxiliary storage tab (property name and origin).

Reply:
But couldn't that be done by re-parsing the field name in a static method too?

Reply:
Yes it's possible to re-parse the field name. 

I think it's more elegant to inherit from QgsField to encapsulate this logic. Moreover, it could be useful in the future if we want to add some more specific behavior to auxiliary field.

However, you're the Jedi here, so if you think it's better to remove QgsAuxiliaryField and use static methods instead, just tell me :)

Reply:
Let's get some more opinions :) @m-kuhn @wonder-sk?

Reply:
With the inheritance approach, is there never a problem of slicing?

```
QgsAuxiliaryField field;
fields.append(field);
fields.last(); // will return a QgsField, not a QgsAuxiliaryField
```